myhoard/update_mysql_environment.py

```diff
@@ -13,26 +15,34 @@ def __init__(self, args):
         self.log = logging.getLogger(self.__class__.__name__)
 
     def update(self):
+        # make sure we only update the parameter we need to update i.e., MYSQLD_OPTS
+        key = "MYSQLD_OPTS"  # we only update this environment variable
+        value = ""
+        if self.args.with_bin_log != "true":
+            value += "--disable-log-bin "
+            # If config says slave-preserve-commit-order=ON MySQL would refuse to start if binlog is
+            # disabled. To prevent that from happening ensure preserve commit order is disabled
+            value += "--skip-slave-preserve-commit-order "
+        if self.args.gtid_mode != "true":
+            value += "--gtid-mode=OFF"
```

Comment:
Creating a list of values and producing a string with `" ".join(values)` would be cleaner and avoid the extra whitespace that may sometimes appear in the output.

Reply:
Thanks, fixed.